src/error.rs

```diff
@@ -148,6 +149,9 @@ impl fmt::Display for TiffUnsupportedError {
             UnsupportedSampleDepth(samples) => {
                 write!(fmt, "{} samples per pixel is unsupported.", samples)
             }
+            UnsupportedSampleFormat(ref formats) => {
+                write!(fmt, "Sample format {:?} is unsupported.", formats)
```

Comment:
Given that one (however common) possible error is that the formats are *mixed*, could we make an effort of trying to display these formats as decoded tags here?

Reply:
The error contains the decoded `SampleFormat` now (including unknown variant), so this should contain all the relevant information.